extensions/window/init.lua

```diff
@@ -682,21 +681,28 @@ end
 ---    - an `hs.geometry` unit rect, or argument to construct one; will set the window frame to this rect relative to the desired screen;
 ---      if `screen` is nil or omitted, use the screen the window is currently on
 ---  * screen - (optional) An `hs.screen` object or argument for `hs.screen.find`; only valid if `rect` is a unit rect
+---  * ensureInScreenBounds - (optional) if `true`, use `setFrameInScreenBounds()` to ensure the resulting window frame is fully contained within
+---    the window's screen; `duration` is ignored (animation is not supported)
 ---  * duration - (optional) A number containing the number of seconds to animate the transition. Defaults to the value of `hs.window.animationDuration`
 ---
 --- Returns:
 ---  * The `hs.window` object
 function window:move(rect,toScreen,duration)
+  local frame,inBounds
+  if type(toScreen)=='boolean' or (type(toScreen)=='number' and toScreen<20 and duration==nil) then duration=toScreen toScreen=nil end
```

Comment:
As a side effect, yes; it's a crude way to disambiguate between a screen id and a duration.
- `w:move('[0,0,50,50]',5123423)` movesToUnitRect on that screen id, if found (otherwise, current screen)
- `w:move('[0,0,50,50]',10)` slooooowly movesToUnitRect on the window's current screen
- `w:move('[0,0,50,50]',634635,10)` slooooowly movesToUnitRect on that screen id

I'm ofc assuming that screen IDs can't be 0<x<20; that seems pretty unlikely